### PR TITLE
Support debugging with lldb

### PR DESCRIPTION
**Pull request by conanoc**
fix https://github.com/ElementsProject/lightning/issues/1093

Running with lldb cause SIGINT, which makes waitpid(), read()
returns error with errno as EINTR. This patch retry these calls
to ignore EINTR errors.

Tested only on macOS.

### CODE REVIEW

File: ccan/ccan/ptr_valid/ptr_valid.c [hunk 1]
```diff
@@ -161,7 +161,7 @@ static void finish_child(struct ptr_valid_batch *batch)
 {
 	close(batch->to_child);
 	close(batch->from_child);
-	waitpid(batch->child_pid, NULL, 0);
+	while(waitpid(batch->child_pid, NULL, 0) < 0 && errno == EINTR);
```
Comment (ZmnSCPxj) [inline ccan/ccan/ptr_valid/ptr_valid.c]: Unfortunately changes to CCAN could should preferably be pushed upstream to CCAN: https://github.com/rustyrussell/ccan

**Comment by arowser** [comment]
Tested pass on macOS 10.12.4

**Comment by rustyrussell** [comment]
Correct fix is to get rid of ptr_valid which is needed by autodata on non-ELF platforms (ie. MacOS).

Ideally someone who knows macos would come up with a clever way of doing autodata there?  Otherwise we could generate the type_to_string and json_command arrays using grep...

**Comment by conanoc** [comment]
Removed other changes except bitcoind.c from the previous commit.

**Comment by conanoc** [comment]
@ZmnSCPxj @rustyrussell I think this patch is ready. Please take a look.

**Comment by ZmnSCPxj** [comment]
ACK 474cc1b24d726d40b422c3277e37aece560e3ddc

**Comment by cdecker** [comment]
ACK 474cc1b

**Pull request merged**